audit/pkg/types/conversion.go

```diff
@@ -231,6 +320,22 @@ func extractGrantee(uid *user.UserId, gid *group.GroupId) (string, string) {
 	return "", ""
 }
 
+func extractFileDetails(ref *provider.Reference, owner *user.UserId) (string, string, string) {
+	id, path := "", ""
+	if ref != nil {
+		path = ref.GetPath()
+		if rid := ref.GetResourceId(); rid != nil {
+			id = rid.GetStorageId() + "!" + rid.GetOpaqueId() + "!" + path
```

Comment:
Our usual spaces reference in string form looks like this: `storageid!opaqueid/path` where path is a relative path without `./` at the beginning. I think it would be nice to use the same layout here for consistency.

Reply:
There is a method `utils.FormatSpacesReference` which you could use for that.

Reply:
Of course you are right. But it is called `utils.FormatStorageSpaceReference` :stuck_out_tongue:

Reply:
Uh, of course. :see_no_evil: I should've looked it up but you've found it anyways. :D